wavefront_pyformance/wavefront_reporter.py

```diff
@@ -24,50 +18,41 @@ class WavefrontReporter(reporter.Reporter):
     # pylint: disable=too-many-arguments
     def __init__(self, source='wavefront-pyformance', registry=None,
                  reporting_interval=10, clock=None, prefix='', tags=None):
-        """Run parent class __init__ and do Wavefront specific set up."""
+        """Construct Wavefront Reporter."""
         super(WavefrontReporter, self).__init__(
-            registry=registry,
-            reporting_interval=reporting_interval,
+            registry=registry, reporting_interval=reporting_interval,
             clock=clock)
+        self.wavefront_client = None
         self.source = source
         self.prefix = prefix
-        tags = tags or {}
-        # pylint: disable=invalid-name
-        self.tagStr = ' '.join('"%s"="%s"' % (k, v) for k, v in tags.items())
+        self.tags = tags or {}
 
     def report_now(self, registry=None, timestamp=None):
-        """Raise error if method is not implemented by derived class."""
-        raise NotImplementedError('use WavefrontProxyReporter'
-                                  ' or WavefrontDirectReporter')
-
-    def _collect_metrics(self, registry, timestamp=None):
+        """Collect metrics from registry and report them to Wavefront."""
+        timestamp = timestamp or int(round(self.clock.time()))
```

Comment:
For the proxy it's okay to set the timestamp here as proxy handles clock drift. For direct we made a decision to not set the timestamp on the client side and let the server handle it when it receives the points. Moving this into the base class enforces setting the timestamp for both.

Let's change this so that the base reporter does not set the timestamp. The proxy reporter can set it and direct will just pass in None.